lld/ELF/InputFiles.cpp

```diff
@@ -194,6 +194,10 @@ static void updateSupportedARMFeatures(const ARMAttributeParser &attributes) {
   if (arch >= ARMBuildAttrs::CPUArch::v8_M_Base &&
       profile == ARMBuildAttrs::MicroControllerProfile)
     config->armCMSESupport = true;
+
+  // The Cortex-M processors only support Thumb.
```

Comment:
As mentioned earlier you may want to be a bit more selective:
* v6-m and v8-m.base can't realistically support PLTs without a very complex instruction sequence.
* Thumb-2 PLTs can be useful outside M profile.